src/wordladder.cpp

```diff
@@ -1,116 +1,307 @@
+    if (wordOne == wordTwo) {
+        cout << "The two words must be different" << endl;
+        return false;
+    }
+    return true;
+}
+
+/*
+ * Function: getWordLadder
+ * Usage:
+ * Find the shortest word ladder from wordOne to wordTwo
+ * Params: string, string
+ * -------------------------------------------------------------------------------------------------
+ * Returns: None. Void function
+*/
+
+void getWordLadder(const Lexicon &dictionary, string wordOne, string wordTwo) {
+    // Initialise a stack storing word 1 only
+    Stack<string> stack {wordOne};
+    // Initialise a queue of stacks storing the initial stack
+    Queue<Stack<string>> queue {stack};
+    // Initialise a set of words use in any previous ladders so these can be ignored to avoid
+    // reusing words
+    Set<string> usedWords {wordOne};
+
+    // Repeat the following until the queue is empty
+    while (!queue.isEmpty()) {
+        // Stop if word 2 is found
+        if (queue.peek().peek() == wordTwo) {
+            break;
+        }
+        // Get the stack containing the first ladder at the front of the queue
+        Stack<string> firstLadder = queue.dequeue();
+        // Find the neighbouring words for the last word on the first Ladder
+        findNeighbourWords(dictionary, usedWords, queue, firstLadder);
+    }
+    // Get the shortest word ladder at the front of the queue
+    if (!queue.isEmpty()) {
+        Stack<string> wordLadderStack = queue.dequeue();
+        // Display the word ladder from word 1 to word 2
+        string wordLadder;
+        for (string word: wordLadderStack) {
+            wordLadder.append(wordLadderStack.pop() + " ");
+        }
+        cout << "A ladder from " << wordTwo << " back to " << wordOne << ":" << endl;
+        cout <<  wordLadder << endl;
+    } else {
+        // If no ladder is found, notify the user
+        cout << "No word ladder found from " << wordTwo << " back to " << wordOne << endl;
+    }
+}
+
+/*
+ * Function: findNeighbourWords
+ * Usage:
+ * Find all the neighbour words from a given start word, which is a word of the same length but
+ * differs from the start word by exactly 1 letter e.g. date and data
+ * Params: dictionary (Lexicon), usedWords (Set), queue (Queue of Stacks of strings),
+ * firstLadder (Stack of strings)
+ * -------------------------------------------------------------------------------------------------
+ * Returns: None. Void function
+*/
+void findNeighbourWords(
+        const Lexicon &dictionary
+        , Set<string> &usedWords
+        , Queue<Stack<string>> &queue
+        , const Stack<string> &firstLadder) {
+    // Get the start word at the top of the initial word ladder
+    string startWord = firstLadder.peek();
+    // The algorith uses 2 nested loops for efficiency: 1st loop for index of the start word,
+    // 2nd loop to loop through a-z of the alphabet
+    for (int i = 0; i < startWord.length(); i++) {
+        for (char l = 'a'; l<='z'; ++l) {

```

Comment:
Instead of `l`, how about `ch` or `letter`? See https://web.stanford.edu/class/archive/cs/cs106b/cs106b.1182/handouts/styleguide.html#naming:

> **Names**: Give variables descriptive names, such as **firstName** or **homeworkScore**. Avoid one-letter names like **x** or **c**, except for loop counter variables such as **i**.

Reply:
Updated!